### DISCUSSION TITLE
Empty vlan IDs resulting in confusing error messages when using vxlan.vlans

### DISCUSSION THREAD

**Discussion by jbemmel** [ipspace/netlab#417] [General]
What triggered me to create https://github.com/ipspace/netlab/pull/405 was an annoying error message when trying to use vxlan.vlans [as documented](https://github.com/ipspace/netlab/blob/dev/docs/module/vxlan.md#selecting-vxlan-enabled-vlans)

```
jeroen@jvm:~/srlinux/netsim-examples/BGP/EVPN-VXLAN-to-hosts$ netlab create
IncorrectType in evpn: attribute nodes.leaf2a.vlans.internet.evpn.evi must be an integer, found dictionary
IncorrectType in evpn: attribute nodes.leaf2a.vlans.openstack-external.evpn.evi must be an integer, found dictionary
...
Fatal error in netlab: Cannot proceed beyond this point due to errors, exiting
```

This happens because those nodes don't have any vlans on them, but the evpn module still expects to use the vlan id

I guess this may be an error in the way vlan ids are generated/assigned, it should be done once globally independent of a vlan being present on a given node

**Comment by ipspace** [comment]
I agree that such an error should not have happened.

If you were to share a minimum topology that results in this annoying error message, then I might be able to figure out what was going on, and fix it.

**Comment by ipspace** [comment]
The VXLAN rewrite might have fixed that. Also, we never got the minimum topology, so nothing to work with :(

**Reply by jbemmel** [reply]
Yes, this seems fixed now
